router-tests/modules/verify-operation-context-values/module.go

```diff
@@ -0,0 +1,85 @@
+package verify_operation_context_values
+
+import (
+	"net/http"
+
+	"github.com/wundergraph/astjson"
+	"github.com/wundergraph/cosmo/router/core"
+	"go.uber.org/zap"
+)
+
+const myModuleID = "verifyOperationContextValues"
+
+// CapturedOperationValues holds the captured values from operation context
+type CapturedOperationValues struct {
+	Name      string
+	Type      string
+	Hash      uint64
+	Content   string
+	Variables *astjson.Value
+	// Store the raw variables as string for easier testing
+	VariablesJSON string
+	ClientInfo    core.ClientInfo
+}
+
+// VerifyOperationContextValuesModule captures operation context values for verification
+type VerifyOperationContextValuesModule struct {
+	ResultsChan chan CapturedOperationValues
+	Logger      *zap.Logger
+}
+
+func (m *VerifyOperationContextValuesModule) Provision(ctx *core.ModuleContext) error {
+	m.Logger = ctx.Logger
+	if m.ResultsChan == nil {
+		m.ResultsChan = make(chan CapturedOperationValues, 1)
+	}
+	return nil
+}
+
+func (m *VerifyOperationContextValuesModule) Middleware(ctx core.RequestContext, next http.Handler) {
+	operation := ctx.Operation()
+
+	// Capture all the operation context values
+	captured := CapturedOperationValues{
+		Name:       operation.Name(),
+		Type:       operation.Type(),
+		Hash:       operation.Hash(),
+		Content:    operation.Content(),
+		Variables:  operation.Variables(),
+		ClientInfo: operation.ClientInfo(),
+	}
+
+	// Convert variables to JSON string for easier testing
+	if captured.Variables != nil {
+		variablesBytes := captured.Variables.MarshalTo(nil)
+		captured.VariablesJSON = string(variablesBytes)
+	} else {
+		captured.VariablesJSON = "{}"
+	}
```

Comment:
Nit: else not needed
```suggestion
        captured.VariablesJSON = "{}"
	if captured.Variables != nil {
		variablesBytes := captured.Variables.MarshalTo(nil)
		captured.VariablesJSON = string(variablesBytes)
	} 
```